components/app-core/backend/main.go

```diff
@@ -589,3 +593,71 @@ func getStaticFiles() (string, error) {
 	}
 	return "", errors.New("UI folder not found")
 }
+
+func isConsoleUpgrading() bool {
+
+	upgradeVolume, noUpgradeVolumeErr := config.GetValue("UPGRADE_VOLUME")
+	upgradeLockFile, noUpgradeLockFileNameErr := config.GetValue("UPGRADE_LOCK_FILENAME")
+
+	// If any of those properties are not set, consider Console is running in a non-upgradeable environment
+	if noUpgradeVolumeErr != nil || noUpgradeLockFileNameErr != nil {
+		return false
+	}
+
+	upgradeLockPath := fmt.Sprintf("/%s/%s", upgradeVolume, upgradeLockFile)
+
+	if _, err := os.Stat(upgradeLockPath); err == nil {
+		return true
+	}
+	return false
+}
+
+func startEchoForUpgrade(config interfaces.PortalConfig) {
```

Comment:
Can we refactor the start func we have and re-use here?

Seems like we could have a shared func that takes a param which determines whether we add the setup middlweware etc (ie. the differences between the two funcs we have).

The "go stopEchoWhenUpgrading" can then be done in the case of upgrade.